arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkippedSection.java

```diff
@@ -35,60 +35,61 @@
  * proving of a pure transaction.
  */
 public class TxSkippedSection extends TraceSection implements PostTransactionDefer {
+
   final TransactionProcessingMetadata txMetadata;
   final AccountSnapshot senderAccountSnapshotBefore;
   final AccountSnapshot recipientAccountSnapshotBefore;
   final AccountSnapshot coinbaseAccountSnapshotBefore;
 
   public TxSkippedSection(
-      Hub hub, WorldView world, TransactionProcessingMetadata txMetadata, Transients transients) {
+      Hub hub,
+      WorldView world,
+      TransactionProcessingMetadata transactionProcessingMetadata,
+      Transients transients) {
     super(hub, (short) 4);
-    this.txMetadata = txMetadata;
     hub.defers().scheduleForPostTransaction(this);
 
-    // From account information
+    txMetadata = transactionProcessingMetadata;
     final Address senderAddress = txMetadata.getBesuTransaction().getSender();
-    senderAccountSnapshotBefore =
-        AccountSnapshot.fromAccount(
-            world.get(senderAddress),
-            isPrecompile(senderAddress),
-            hub.deploymentNumberOf(senderAddress),
-            false);
-
-    // Recipiet account information
     final Address recipientAddress = txMetadata.getEffectiveRecipient();
-    recipientAccountSnapshotBefore =
-        AccountSnapshot.fromAccount(
-            world.get(recipientAddress),
-            isPrecompile(recipientAddress),
-            hub.deploymentNumberOf(recipientAddress),
-            false);
+    final Address coinbaseAddress = txMetadata.getCoinbase();
+
+    senderAccountSnapshotBefore = AccountSnapshot.canonical(hub, senderAddress);
+    recipientAccountSnapshotBefore = AccountSnapshot.canonical(hub, recipientAddress);
+    coinbaseAccountSnapshotBefore = AccountSnapshot.canonical(hub, coinbaseAddress);
+
+    // sanity check + EIP-3607
+    checkArgument(world.get(senderAddress) != null);
+    checkArgument(!world.get(senderAddress).hasCode());
+
+    // deployments are local to a transaction, every address should have deploymentStatus == false
+    // at the start of every transaction
+    checkArgument(!hub.deploymentStatusOf(senderAddress));
```

Comment:
I don't like the name `deploymentStatus` : it sounds like the output of a status is an enum, not a boolean. I'd prefer `underDeployment` / `deployement` / `!deployed`

Reply:
The whole spec uses that name ...